### PR TITLE
Added special attributes to event-creation notifications

### PR DESCRIPTION
**Pull request by Johannes-Schneider**

### CODE REVIEW

File: src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpoint.java [hunk 1]
```diff
@@ -144,16 +144,16 @@ static String getEventTypesOfEntityUri(long entityId, boolean includeChildEvents
     /**
      * This method returns the basic URI to retrieve the events of an entity.
      * @param entityId - the id of the entity to be searched for
-     * @param entityTypeId - the id of the entity type to be searched for
+     * @param eventTypeId - the id of the entity type to be searched for
```
Comment (frederike-ramin) [inline src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpoint.java]: it still says entity type in the comment.

File: src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapter.java [hunk 2]
```diff
@@ -49,6 +49,14 @@
 	boolean createArtifact(PersistenceArtifact artifact, String fetchUri);
 
 	/**
+	 * This method saves a new event in the database and notifies connected web socket clients.
+	 * @param event
+	 * @param eventOwner
+	 * @return
+	 */
+	boolean createEvent(Event event, Entity eventOwner, String fetchUri);
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapter.java]: MAJOR Expected an @return tag. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadocMethod_BP2016)
MAJOR Expected @param tag for 'event'. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadocMethod_BP2016)
MAJOR Expected @param tag for 'eventOwner'. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadocMethod_BP2016)
MAJOR Expected @param tag for 'fetchUri'. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadocMethod_BP2016)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/test/java/de/hpi/bpt/argos/eventProcessing/EventReceiverTest.java [hunk 3]
```diff
@@ -77,6 +77,11 @@ public void testReceiveEvent() throws Exception {
 
 		List<String> webSocketMessages = webSocket.awaitMessages(1, 1000);
 		ArgosTestUtil.assertWebSocketMessage(webSocketMessages.get(0), PersistenceArtifactUpdateType.CREATE, "Event");
+
+		JsonObject message = jsonParser.parse(webSocketMessages.get(0)).getAsJsonArray().get(0).getAsJsonObject();
```
Comment (frederike-ramin) [inline src/test/java/de/hpi/bpt/argos/eventProcessing/EventReceiverTest.java:81]: Why don't you use the assertWebSocketMessage metod here too?

**Review by frederike-ramin — CHANGES_REQUESTED** [review]

File: src/test/java/de/hpi/bpt/argos/eventProcessing/EventReceiverTest.java (hunk 3 above)
Comment (Johannes-Schneider) [reply src/test/java/de/hpi/bpt/argos/eventProcessing/EventReceiverTest.java:81]: This method asserts two messages. One for a created event and one for an entity update. In this case, the status of the entity should not be updated and therefore only one message is to be asserted.

**Review by frederike-ramin — APPROVED** [review]

File: src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapter.java [hunk 4]
```diff
@@ -49,6 +49,14 @@
 	boolean createArtifact(PersistenceArtifact artifact, String fetchUri);
 
 	/**
+	 * This method saves a new event in the database and notifies connected web socket clients.
+	 * @param event - the created event
+	 * @param eventOwner - the entity, which owns the event
+	 * @return - true, if the event was stored in the database
+	 */
+	boolean createEvent(Event event, Entity eventOwner, String fetchUri);
```
Comment (hpibptlab) [inline src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapter.java]: MAJOR Expected @param tag for 'fetchUri'. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AJavadocMethod_BP2016)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Pull request merged**